### DISCUSSION TITLE
New Architecture is now in Beta!

### DISCUSSION THREAD

**Discussion by cipolleschi** [reactwg/react-native-new-architecture#189] [Announcements]
_TL;DR: Starting today, React Native’s New Architecture is now in Beta. We believe the experience is stable enough for you to test in your application and we need your help tracking down any remaining bugs. We’re expecting to release the New Architecture as stable by the end of 2024._

Hi all,
Today, we are excited to promote the **New Architecture** from Experimental **to Beta**! 🎉

Over the last few months we’ve fixed many New Architecture bugs and we believe the experience is now close to stable. Moreover, most of the [most popular React Native libraries](https://github.com/reactwg/react-native-new-architecture/discussions/167) have been migrated to fully support the New Architecture. 

We’re asking React Native developers to try the New Architecture in their apps, and report back on your experience and any bugs you run into. As this is a Beta, we do still expect there to be some bugs or other issues. With your help we can get these resolved and get the New Architecture to a stable release.

We expect to have a stable release with the New Architecture by the end of 2024.

### Expo
We believe that the best way to experience the New Architecture is through Expo SDK 51.

We collaborated closely with Expo to make sure that all the Expo modules shipped in SDK 51 are working seamlessly in apps where the New Architecture is turned on.

To turn on the New Architecture with Expo, follow the instructions [at this page](https://docs.expo.dev/guides/new-architecture/#enable-the-new-architecture-in-an-existing-project).

### React Native projects without a framework
If you are not using Expo, you need to turn on the New Architecture manually in your iOS and Android applications.
First, migrate to the latest version of React Native and then follow the instructions in [this guide](https://github.com/reactwg/react-native-new-architecture/blob/main/docs/enable-apps.md).

#### Android

1. Open the `<your-app>/android/gradle.properties` file
2. Turn on the `newArchEnabled` flag
	```diff
	# Use this property to enable support for the new architecture.
    # This will allow you to use TurboModules and the Fabric render in
    # your application. You should enable this flag either if you want
    # to write custom TurboModules/Fabric components OR use libraries that
    # are providing them.
    - newArchEnabled=false
    + newArchEnabled=true
	```

#### iOS

1. Navigate to your `ios` folder from a terminal `cd <your-app>/ios`
2. Install your dependencies with the `RCT_NEW_ARCH_ENABLED=1`: `bundle install && RCT_NEW_ARCH_ENABLED=1 bundle exec pod install`

## What does this mean for app developers?

Over the last months, we invested in making the New Architecture migration easier and smoother for as many developers as possible.

Specifically in React Native 0.74, we released more [Backward Compatibility APIs](https://github.com/reactwg/react-native-new-architecture/discussions/174#backward-compatibility) and [Interoperability Layers](https://github.com/reactwg/react-native-new-architecture/discussions/175).

On top of this, during the release of React Native 0.74, we collaborated with Expo, Software Mansion, Callstack and several other community libraries to [fully support the New Architecture](https://github.com/reactwg/react-native-new-architecture/discussions/167).

With those changes, we believe you should be able to just turn on the New Architecture **without changing a single line of code**. In most cases, your apps should just work, but more complex applications might require further migration steps. If so, we’re interested in learning more about your migration story (see below).

## What does this mean for Library developers?

If you are a library maintainer, we are asking you to try your library with the New Architecture turned on.

Even if you have not migrated your library yet, it will go through the Interoperability Layer and we expect for it to work.

If you are encountering issues with the interoperability layers, please report them using [this issue template](https://github.com/facebook/react-native/issues/new?assignees=&labels=Needs%3A+Triage+%3Amag%3A%2CType%3A+New+Architecture&projects=&template=new_architecture_bug_report.yml). 

> [!WARNING]
> Complex libraries might not work fully with the interoperability layer. If that’s the case, you might need to migrate your library to the New Architecture to make it work properly.
> If that’s the case and you need help, please reach out in the Working Group so we can find a strategy to move forward together.

### How to report issues?

> [!NOTE]
> Please note that the New Architecture is still in Beta. Please report any bugs or problems you notice when turning on the New Architecture.

If that happens in your app, please report those issues using [this issue template](https://github.com/facebook/react-native/issues/new?assignees=&labels=Needs%3A+Triage+%3Amag%3A%2CType%3A+New+Architecture&projects=&template=new_architecture_bug_report.yml). 

This working group has been crucial in shaping the New Architecture rollout, and we hope you’ll help us make us through the final line! 🚀
